tests/utils.py

```diff
@@ -511,6 +511,25 @@ def tearDown(self):
         self.loop.close()
 
 
+class NoAuthUserServerTestCase(unittest.TestCase):
```

Comment:
Since you use pytest to run tests anyway, I think it would be better to use pytest fixtures instead of unittest-style hooks.

```python
@pytest.fixture
def no_auth_user_server():
    server = NATSD(...)
    yield server
    server.stop()
    
```